src/Runner.Listener/MessageListener.cs

```diff
@@ -268,6 +268,7 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
                 }
 
                 Trace.Info($"Message '{message.MessageId}' received from session '{_session.SessionId}'.");
+                Trace.Info($"Message '{message.MessageId}' char length: '{message.Body.Length}', byte length '{System.Text.ASCIIEncoding.Unicode.GetByteCount(message.Body)}'.");
```

Comment:
i think we want to do this tracing in src/Runner.Common/ProcessChannel.cs instead in SendAsync and ReceiveAsync

Reply:
thinking.... instead of tracing byte length can we trace the string length and a hash of the contents

That should tell us whether the string got malformed in the process. For instance if any codepoints didnt roundtrip properly

Reply:
Actually maybe not bad to go ahead and trace the string length here.

But you definitely want to trace in the processchannel too. The reason is, message.Body isn't what get's sent to the worker process. It gets deserialized and serialized again before being sent via processchannel.

Reply:
👋  Thanks for the feedback! That's a good point about tracing in the process channel since that's closer to the failure point in the scenario that this trace would be used for. I've gone ahead and done that and left the other traces in. If you think they're valuable let's keep them, if not let's yank them out.